cs/src/core/Index/FASTER/FASTERThread.cs

```diff
@@ -27,7 +27,7 @@ public partial class FasterKV<Key, Value> : FasterBase, IFasterKV<Key, Value>
                     var currentState = SystemState.Copy(ref systemState);
                     if (currentState.phase == Phase.REST)
                     {
-                        var intermediateState = SystemState.Make(Phase.INTERMEDIATE, currentState.version);
+                        var intermediateState = SystemState.Make(currentState.phase | Phase.INTERMEDIATE, currentState.version);
```

Comment:
Maybe refactor this into a method for readability?

Reply:
Done